src/terminal/message.rs

```diff
@@ -1,9 +1,55 @@
 use super::emoji;
+use once_cell::sync::OnceCell;
 
 use billboard::{Billboard, BorderColor, BorderStyle};
+use serde::{Deserialize, Serialize};
+
+#[derive(PartialEq)]
+pub enum OutputType {
+    Json,
+    Human,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct PublishOutput {
+    pub success: Option<bool>,
+    pub name: Option<String>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub urls: Vec<String>,
+}
+
+static OUTPUT_TYPE: OnceCell<OutputType> = OnceCell::new();
+
+pub fn set_output_type(typ: OutputType) {
+    match OUTPUT_TYPE.set(typ) {
+        Ok(_) => {}
+        Err(_) => {
+            message(&"Output type already set".to_string());
```

Comment:
probably dont need to message here if this is the case, can do a log::info if you want though

Reply:
Would it be ok to just `.unwrap()` the value here? Not sure what the rust convention is when you don't care about a `Result`.

Reply:
no - unwrapping an error will cause it to panic. if you want to ignore you can do this

```rust
let _ = OUTPUT_TYPE.set(r#type);
```